etcdserver/apply.go

```diff
@@ -53,7 +53,7 @@ type applierV3 interface {
 	Put(txnID int64, p *pb.PutRequest) (*pb.PutResponse, error)
 	Range(txnID int64, r *pb.RangeRequest) (*pb.RangeResponse, error)
 	DeleteRange(txnID int64, dr *pb.DeleteRangeRequest) (*pb.DeleteRangeResponse, error)
-	Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error)
+	Txn(header *pb.RequestHeader, rt *pb.TxnRequest) (*pb.TxnResponse, error)
```

Comment:
I don't think the auth checking should be smeared between `applierV3` and `applyV3Request` like this since it blurs the line of responsibility. There probably should be an authApplierV3 interface which adds an auth argument to the relevant applierV3 methods; the implementation would pass the request down to an underlying `applierV3` call if auth checks pass.